numpy/lib/_function_base_impl.py

```diff
@@ -2332,7 +2332,7 @@ class vectorize:
 
     cache : bool, optional
         If `True`, then cache the first function call that determines the number
-        of outputs if `otypes` is not provided.
+        of outputs if `otypes` and `signature` are not provided.
```

Comment:
Where in the code does `signature` come into play? I took a cursory look and it seemed only `otypes` is involved in the cache. In any case, perhaps:

If ``True``, then cache the number of outputs from a function call in order to speed up subsequent calls. Effective only if `otypes` is not provided (or Effective only if neither `otypes` nor `signature` are provided).

Reply:
Here: https://github.com/numpy/numpy/blob/d621a3162e7a14e986e2dee29bbf9ff05f7b728d/numpy/lib/_function_base_impl.py#L2622

The cache-related logic isn't even reached if `signature` is provided.

About the wording, could we tackle that separately?

Reply:
OK, so the change seems correct. 
> could we tackle that separately?

@rkern wrote the original docstring, maybe he has an opinion.

Reply:
I did? I don't really know any of the details now, at least; I don't really use `vectorize` and am unfamiliar with the caching functionality.

Reply:
Could this be merged then?